Sources/GRPCNIOTransportCore/Client/Connection/LoadBalancers/Subchannel.swift

```diff
@@ -282,16 +282,27 @@ extension Subchannel {
     }
   }
 
-  private func handleConnectFailedEvent(in group: inout DiscardingTaskGroup) {
+  private func handleConnectFailedEvent(in group: inout DiscardingTaskGroup, error: any Error) {
```

Comment:
Rather than `any Error`, can we use `RPCError` and propagate it back to wherever the `connectFailed` event came from?

Reply:
Okay I had to do some gymnastics to get this done - not sure if it's compiler errors or I'm misunderstanding something about how Result deals with typed throws, but it wasn't too straightforward 😅